solc_json_parser/parser.py

```diff
@@ -814,13 +814,20 @@ def __parse_asm_data(self, contract_name, deploy=False) -> Dict[str, Any]:
     def get_source_list(self):
         source_list = []
         idx2path = {}
+        yul_support_flag = False
         for contract_name in self.solc_json_ast.keys():
             absolute_path = self.source_path_by_contract(contract_name)
             idx = get_in(self.solc_json_ast, contract_name, 'ast', 'src').split(':')[-1]
             idx2path[int(idx)] = absolute_path
+            if self.solc_json_ast.get(contract_name).get('generated-sources') and \
+                    self.solc_json_ast.get(contract_name).get('generated-sources-runtime'):
+                yul_support_flag = True
         sort_keys = sorted(idx2path.keys())
         for idx in sort_keys:
             source_list.append(idx2path[idx])
+
+        if yul_support_flag:
+            source_list.append("#utility.yul")
```

Comment:
1. Some documentation would be nice here, especially for `yul_support_flag` 
2. any meaning for the hardcoded `#utility.yul`?

Reply:
I will update will some comments on this.

Reply:
`contract.get('asm').get('sourceList')` is introduced very lately, maybe around 0.8.15. In this sourceList we get from the output, "#utility.yul" is at last. For example, if there are two files, a.sol and b.sol, the sourceList will look like:
["a.sol", "b.sol", "#utility.yul"]. But in the earlier version, we get the sourceList by analyzing the AST. Yul is introduced in solidity in 0.7.2. So between 0.7.2 and 0.8.15, I want to get a same output from `self.get_source_list()`. So I check if there are `generated-sources` and `generated-sources-runtime` in it by using the `yul_support_flag`.